Makefile.menhir

```diff
@@ -76,7 +76,7 @@ MENHIRBASICFLAGS := \
 MENHIRFLAGS := \
   $(MENHIRBASICFLAGS) \
   --infer \
-  --ocamlc "$(CAMLC) $(COMPFLAGS)" \
+  --ocamlc "$(CAMLC) $(OC_COMMON_CFLAGS) $(INCLUDES)" \
```

Comment:
I am not sure if we really want to use the common ocamlc flags for menhir inference: it makes sense to be as lax as possible here since menhir is not using `ocamlc` to compile some source file but just trying to get back some type information.

Reply:
That can certainly be tweaked - the PR here restored the behaviour pre-#11420 (which effectively renamed `COMPFLAGS` in the root flags to `OC_COMMON_CFLAGS` minus the `INCLUDES` part).